templates/profile/prometheus/rules.yml.erb

```diff
@@ -353,6 +353,24 @@ groups:
     annotations:
       summary: 'MySQL Connection Count is approaching MAX on {{$labels.hostname}}'
       description: 'MySQL Connection Count is approaching MAX'
+  - alert: "EtcdDatabaseFillingUpFast"
+    expr: >
+      max by (datacenter) (delta(etcd_debugging_mvcc_db_total_size_in_bytes[5m])) > 10000
```

Comment:
This means consistently +10KB every 5mins for 30 minutes, right? So, I think, as written with delta:

- Bursts would not trigger this threshold (I think delta 500MB over 20 minutes would fall to 0 within the window)
- The alert slope may be too low / noisy (it seems like 60KB would be quite easy to hit with CRDs and CI/CD)

I'm not sure, but this seems like a case for a subquery (or `deriv`):

https://prometheus.io/blog/2019/01/28/subquery-support/#motivation
https://prometheus.io/docs/prometheus/latest/querying/examples/#subquery
https://prometheus.io/docs/prometheus/latest/querying/functions/#deriv
https://grafana.com/blog/2019/04/01/how-were-using-prometheus-subqueries-at-grafana-labs./

What do you think?

Reply:
![Screenshot from 2025-05-02 11-19-53](https://github.com/user-attachments/assets/7eb08a3c-81e1-4024-9dcb-7580c80a798a)

All I did was find a threshold noisy enough to pick up clusters that weren't having trouble but quiet enough that `for: 30m` addresses that.

Reply:
Which is to say, when this was happening to us in real time, the delta was greater than 1Mi, constantly, for a period of 15 hours, so a threshold of 10Ki for 0.5 hours should start firing alerts if this happens again.

Reply:
Huh, can't argue with that graph. Ship it!